### PR TITLE
Native workalike to `compute_tree_root` in `plonk::stlib::merkle_tree`

### PR DESCRIPTION
**Pull request by dbanks12**
# Description

Native workalike to `compute_tree_root` in `plonk::stlib::merkle_tree`

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [x] There are no circuit changes, OR a cryptographer has been assigned for review.
- [x] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [ ] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.
- [x] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [x] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

**Comment by suyash67** [comment]
We need to call another function if we're using UltraPlonk. So you'll have to add conditional logic like:
```c++
if (plonk::SYSTEM_COMPOSER == plonk::PLOOKUP) {
    return crypto::pedersen_hash::lookup::hash_multiple({ lhs, rhs });
} else {
    return crypto::pedersen_hash::hash_multiple({ lhs, rhs });
}
```

**Comment by dbanks12** [comment]
> We need to call another function if we're using UltraPlonk. So you'll have to add conditional logic like:

Thanks for the feedback @suyash67! Why do we need to use different pedersen implementations here when these are native functions? What makes a **native** pedersen hash different when done in via ultraplonk?

**Comment by dbanks12** [comment]
Fixed here: https://github.com/AztecProtocol/barretenberg/pull/247/commits/c379a74ad0d80ea36b3738ecb1a76b5d78b61bdd

**Comment by suyash67** [comment]
> > We need to call another function if we're using UltraPlonk. So you'll have to add conditional logic like:
> 
> Thanks for the feedback @suyash67! Why do we need to use different pedersen implementations here when these are native functions? What makes a **native** pedersen hash different when done in via ultraplonk?

Sorry for a late reply, I missed this comment. Basically, ultraplonk uses a very efficient lookup based Pedersen hash for merkle tree hashing. On the other hand, Pedersen hash using turbo plonk uses a custom fixed base gate. The output of the Pedersen when using turbo and ultra are different (because the underlying designs are totally different). That's the reason we want the conditional logic.

File: cpp/src/barretenberg/stdlib/merkle_tree/hash.hpp [hunk 1]
```diff
@@ -22,6 +22,34 @@ inline barretenberg::fr compress_native(barretenberg::fr const& lhs, barretenber
...
+inline barretenberg::fr compute_tree_root_native(std::vector<barretenberg::fr> const& input)
+{
+    // Check if the input vector size is a power of 2.
+    ASSERT(input.size() > 0);
+    ASSERT(!(input.size() & (input.size() - 1)) == true);
+    auto layer = input;
+    while (layer.size() > 1) {
+        std::vector<barretenberg::fr> next_layer(layer.size() / 2);
+        for (size_t i = 0; i < next_layer.size(); ++i) {
+            next_layer[i] = crypto::pedersen_hash::hash_multiple({ layer[i * 2], layer[i * 2 + 1] });
```
Comment (suyash67) [inline cpp/src/barretenberg/stdlib/merkle_tree/hash.hpp]: Do we need this line?

**Pull request merged**